fix: backend /docs returns 500 causing smoke test failure after deploy

## Summary

After the deploy pipeline successfully provisions both `BackendLambdaStack` and `StaticSiteStack`, the post-deploy smoke test fails because the backend `/docs` endpoint returns HTTP 500.

## Failing step

**CI run:** https://github.com/leonarduk/allotmint/actions/runs/25287288615/job/74133627580 *(Deploy Infrastructure #4885)*

```
curl --fail --show-error --silent "${backend_url%/}/docs" >/dev/null
curl: (22) The requested URL returned error: 500
Error: Process completed with exit code 22.
```

Backend URL at time of failure: `https://dxt34gh2i0.execute-api.***.amazonaws.com`

## Context

- The CDK stacks deploy cleanly (StaticSiteStack reached `CREATE_COMPLETE 22/22`).
- The frontend smoke test curl (CloudFront domain) was not reached — the backend check failed first.
- This started failing after merging PR #2802 (which only touched `frontend/src/pages/Reports.tsx`), so the root cause is likely pre-existing or triggered by a Lambda cold-start/config issue rather than that frontend change.

## Likely causes

- FastAPI app factory raises an unhandled exception on startup (missing env var, failed import, bad config), causing Lambda to return 500 on every request including `/docs`.
- The `/docs` route is explicitly disabled or broken in the Lambda handler configuration.
- A dependency in `requirements.txt` is missing or incompatible in the Lambda runtime environment.

## Investigation starting points

- Check Lambda function logs in CloudWatch for the stack deployed in this run.
- Reproduce locally: `bash scripts/bash/run-local-api.sh` and `curl localhost:<port>/docs`.
- Review `backend/app.py` (`create_app`) for any startup-time side effects that could fail in Lambda context.
- Confirm `/docs` is not explicitly excluded from the Lambda handler routing.

## Acceptance criteria

- [ ] `curl "${BACKEND_URL}/docs"` returns HTTP 200 after a clean deploy.
- [ ] The smoke test step passes end-to-end in CI.
- [ ] Root cause is documented in this issue or the PR description.

--- a/backend/bootstrap/startup.py
+++ b/backend/bootstrap/startup.py
@@ -64,8 +64,11 @@ class AppLifecycleService:
         try:
             await asyncio.to_thread(instrument_api.prime_latest_prices)
         except Exception:
+            # Non-fatal: price priming is a startup optimisation. If it fails (e.g.
+            # network unavailable on Lambda cold start) the app should still serve
+            # requests; a re-raise here propagates through the ASGI lifespan and
+            # causes every subsequent request to return 500.
             logger.exception("Failed to prime latest prices from warmed snapshot")
-            raise
 
     @staticmethod
     def _flush_logging() -> None:

--- a/tests/backend/test_app_bootstrap.py
+++ b/tests/backend/test_app_bootstrap.py
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import asyncio
+import logging
 from pathlib import Path
 
 import pytest
@@ -139,20 +140,19 @@ async def test_lifecycle_service_startup_survives_prime_latest_prices_failure(
         _fail_prime,
     )
     monkeypatch.setattr("backend.bootstrap.startup.refresh_snapshot_async", lambda days: None)
+    monkeypatch.setattr(config, "skip_snapshot_warm", False, raising=False)
 
-    config.skip_snapshot_warm = False
     service = AppLifecycleService(cfg=config)
     app = app_module.create_app()
-
-    import logging
 
     with caplog.at_level(logging.ERROR):
         # Must not raise — previously the re-raise turned this into a 500 for all requests.
         await service.startup(app)
 
-    assert any("prime" in r.message.lower() for r in caplog.records), (
-        "Expected a log record mentioning 'prime' but got: "
-        + str([r.message for r in caplog.records])
+    assert any(
+        "Failed to prime latest prices" in r.message for r in caplog.records
+    ), "Expected 'Failed to prime latest prices' log but got: " + str(
+        [r.message for r in caplog.records]
     )
 
 

--- a/tests/backend/test_app_bootstrap.py
+++ b/tests/backend/test_app_bootstrap.py
@@ -113,6 +113,7 @@ async def test_lifecycle_service_shutdown_cancels_tasks_and_temp_dirs(
 @pytest.mark.asyncio
 async def test_lifecycle_service_startup_survives_prime_latest_prices_failure(
     monkeypatch: pytest.MonkeyPatch,
+    caplog: pytest.LogCaptureFixture,
 ):
     """Startup must not raise when prime_latest_prices fails.
 
@@ -120,6 +121,10 @@ async def test_lifecycle_service_startup_survives_prime_latest_prices_failure(
     request (including /docs) returns 500.  Remove the raise so that Lambda
     cold-start network failures are non-fatal.
     """
+
+    def _fail_prime():
+        raise RuntimeError("simulated network failure")
+
     monkeypatch.setattr("backend.bootstrap.startup._load_snapshot", lambda: ({}, None))
     monkeypatch.setattr(
         "backend.bootstrap.startup.refresh_snapshot_in_memory",
@@ -131,7 +136,7 @@ async def test_lifecycle_service_startup_survives_prime_latest_prices_failure(
     )
     monkeypatch.setattr(
         "backend.common.instrument_api.prime_latest_prices",
-        lambda: (_ for _ in ()).throw(RuntimeError("simulated network failure")),
+        _fail_prime,
     )
     monkeypatch.setattr("backend.bootstrap.startup.refresh_snapshot_async", lambda days: None)
 
@@ -139,8 +144,16 @@ async def test_lifecycle_service_startup_survives_prime_latest_prices_failure(
     service = AppLifecycleService(cfg=config)
     app = app_module.create_app()
 
-    # Must not raise — previously the re-raise turned this into a 500 for all requests.
-    await service.startup(app)
+    import logging
+
+    with caplog.at_level(logging.ERROR):
+        # Must not raise — previously the re-raise turned this into a 500 for all requests.
+        await service.startup(app)
+
+    assert any("prime" in r.message.lower() for r in caplog.records), (
+        "Expected a log record mentioning 'prime' but got: "
+        + str([r.message for r in caplog.records])
+    )
 
 
 def test_create_app_skips_snapshot_warm_when_disabled(monkeypatch: pytest.MonkeyPatch):

--- a/tests/backend/test_app_bootstrap.py
+++ b/tests/backend/test_app_bootstrap.py
@@ -110,6 +110,39 @@ async def test_lifecycle_service_shutdown_cancels_tasks_and_temp_dirs(
     assert not temp_dir.exists()
 
 
+@pytest.mark.asyncio
+async def test_lifecycle_service_startup_survives_prime_latest_prices_failure(
+    monkeypatch: pytest.MonkeyPatch,
+):
+    """Startup must not raise when prime_latest_prices fails.
+
+    If the re-raise is present, the ASGI lifespan aborts and every subsequent
+    request (including /docs) returns 500.  Remove the raise so that Lambda
+    cold-start network failures are non-fatal.
+    """
+    monkeypatch.setattr("backend.bootstrap.startup._load_snapshot", lambda: ({}, None))
+    monkeypatch.setattr(
+        "backend.bootstrap.startup.refresh_snapshot_in_memory",
+        lambda snapshot, ts: None,
+    )
+    monkeypatch.setattr(
+        "backend.common.instrument_api.update_latest_prices_from_snapshot",
+        lambda snapshot: None,
+    )
+    monkeypatch.setattr(
+        "backend.common.instrument_api.prime_latest_prices",
+        lambda: (_ for _ in ()).throw(RuntimeError("simulated network failure")),
+    )
+    monkeypatch.setattr("backend.bootstrap.startup.refresh_snapshot_async", lambda days: None)
+
+    config.skip_snapshot_warm = False
+    service = AppLifecycleService(cfg=config)
+    app = app_module.create_app()
+
+    # Must not raise — previously the re-raise turned this into a 500 for all requests.
+    await service.startup(app)
+
+
 def test_create_app_skips_snapshot_warm_when_disabled(monkeypatch: pytest.MonkeyPatch):
     monkeypatch.setattr(config, "skip_snapshot_warm", True, raising=False)
 

